src/Silverback.Integration/Messaging/Broker/Consumer.cs

```diff
@@ -1,59 +1,114 @@
+
+namespace Silverback.Messaging.Broker
+{
+    public abstract class Consumer : EndpointConnectedObject, IConsumer
+    {
+        private static readonly DiagnosticListener _diagnosticListener = new DiagnosticListener(DiagnosticsConstants.DiagnosticListenerNameConsumer);
+
+        private readonly ILogger<Consumer> _logger;
+        private readonly MessageLogger _messageLogger;
+
+        protected Consumer(IBroker broker, IEndpoint endpoint, ILogger<Consumer> logger, MessageLogger messageLogger)
+           : base(broker, endpoint)
+        {
+            _logger = logger;
+            _messageLogger = messageLogger;
+        }
+
+        public event EventHandler<MessageReceivedEventArgs> Received;
+
+        public void Acknowledge(IOffset offset)
+        {
+            Acknowledge(new[] { offset });
+        }
+
+        public abstract void Acknowledge(IEnumerable<IOffset> offsets);
+
+        protected void HandleMessage(byte[] message, IEnumerable<MessageHeader> headers, IOffset offset)
+        {
+            Activity activity = StartActivity(headers);
+            try
+            {
+                InternalHandleMessage(message, headers, offset);
+            }
+            finally
+            {
+                StopActivity(activity);
+            }
+        }
+
+        private void InternalHandleMessage(byte[] message, IEnumerable<MessageHeader> headers, IOffset offset)
+        {
+            if (Received == null)
+                throw new InvalidOperationException("A message was received but no handler is configured, please attach to the Received event.");
+
+            var deserializedMessage = Endpoint.Serializer.Deserialize(message);
+
+            _messageLogger.LogTrace(_logger, "Message received.", deserializedMessage, Endpoint);
+
+            Received.Invoke(this, new MessageReceivedEventArgs(deserializedMessage, headers, offset));
+        }
+
+        private Activity StartActivity(IEnumerable<MessageHeader> messageHeaders)
+        {
+            string correlationId =  messageHeaders.GetFromHeaders(DiagnosticsConstants.CorrelationIdHeaderKey);
+
+            var activity = new Activity(DiagnosticsConstants.ActivityNameMessageConsuming);
+            if (!string.IsNullOrEmpty(correlationId))
+            {
+                // This will reflect, that the current activity is a child of the activity
+                // which is represented in the message.
+                activity.SetParentId(correlationId);
+
+                // We expect baggage to be empty by default
+                // Only very advanced users will be using it in near future, we encourage them to keep baggage small (few items)
+                string baggage = messageHeaders.GetFromHeaders(DiagnosticsConstants.CorrelationBaggageHeaderKey);
+                if (BaggageConverter.TryDeserialize(baggage, out var baggageItems))
+                {
+                    activity.AddBaggageRange(baggageItems);
+                }
+            }
+

```

Comment:
I saw that the .net guys now let you enable the CorrelationId propagation without using diagnosticsListener.
However, it would be cool to support them anyway, as this will kind of replace ETW.